docs/developer-docs/daos/nns/neuron-management.md

```diff
@@ -0,0 +1,210 @@
+* **Age**: Every non-dissolving neuron has an age, which denotes how long it has been in the non-dissolving state. 
+A neuron's dissolve delay and age increases the neuron's voting power. 
+
+* **Maturity (positive number – percent)**:  When neurons vote, over time the NNS increases their maturity to reward them. Maturity can be converted into ICP by spawning (see below).
+
+## Interacting with neurons
+All interactions with a neuron go through the following API on the NNS governance canister. This includes everything in a neuron's lifetime such as the neuron's creation, modifications and voting, and unstaking the ICP tokens.
+```
+type ManageNeuron = record {
+  id : opt NeuronId;
+  command : opt Command;
+  neuron_id_or_subaccount : opt NeuronIdOrSubaccount;
+};
+```
+
+### Creating a neuron and topping it up
+Staking ICP utility tokens in a neuron involves two commands:
+1. Sending ICP utility tokens to the neuron's subaccount.
+2. Claiming the neuron on the NNS governance canister, which tells the governance canister that the transfer in (1) happened, upon which the NNS governance will create a new neuron.
+
+To follow these steps, you first need to know the principal that should control the neuron and you need to choose a nonce. Based on these two inputs you can compute the required subaccount.
+View for example the method [`compute_neuron_staking_subaccount_bytes`](https://github.com/dfinity/ic/blob/master/rs/nervous_system/common/src/ledger.rs) which computes the subaccount given a controller principal and a nonce. Make sure you are using an up-to-date version of this code in your application.
+
+:::info
+It is of utmost importance to ensure that the subaccount is computed correctly in the first step and remembered so that it can be reused in the second step.
+If this fails, funds could be sent to a dead account and be unrecoverable.
+
+:::
+
+First learn the controller principal and choose a nonce.
+Then, for the ledger transfer, view the [relevant `candid` interface](https://github.com/dfinity/ic/blob/master/rs/rosetta-api/icp_ledger/ledger.did#L25) which contains the following details for a transaction.
+```
+type Transaction = record {
+    memo : Memo;
+    icrc1_memo: opt blob;
+    operation : opt Operation;
+    created_at_time : TimeStamp;
+}; 
+type Operation = variant {
+    ...
+    Transfer : record {
+        from : AccountIdentifier;
+        to : AccountIdentifier;
+        amount : Tokens;
+        fee : Tokens;
+        spender : opt vec nat8;
+    };
+    ...
+};
+```
+To make a transfer to the correct subaccount, choose the `memo` equal your chosen nonce and `operation` to be a transfer. You can make the transfer from any `from` account that you control. The `to` account should be the computed subaccount for the neuron. To later stake a neuron, the `amount` must be at least 1 ICP. Set the `fee` to the standard fee of the ICP ledger canister.
```

Comment:
So the "to" account has to be AccountIdentifier::new(GOVERNANCE_CANISTER_ID.get(), Some(subaccount))

If you look at claim_neuron you can see that neuron_subaccount is used.  This uses the computed subaccount along with Governance to create the account that is checked on the ledger. https://sourcegraph.com/github.com/dfinity/ic@0cdb6f9d14a65906f0089b6571fcdd5b9867d19b/-/blob/rs/nns/governance/src/governance.rs?L5675 

The subaccount is computed in the way you say, but I worry here that someone might try to send funds directly to the the subaccount in this step.  The subaccount is only part of the "to" field for the transfer.  The principal is GOVERNANCE_CANISTER_ID.  

What that actually looks like in bytes would be very different than the subaccount.

Reply:
Andre looked at this in the meantime so I am resolving the thread